packages/merge-styles/src/transforms/kebabRules.ts

```diff
@@ -1,124 +1,16 @@
-  borderTopLeftRadius: 'border-top-left-radius',
-  borderTopRightRadius: 'border-top-right-radius',
-  borderTopStyle: 'border-top-style',
-  borderTopWidth: 'border-top-width',
-  borderWidth: 'border-width',
-  bordercolor: 'bordercolor',
-  bottom: 'bottom',
-  boxShadow: 'box-shadow',
-  boxSizing: 'box-sizing',
-  clear: 'clear',
-  color: 'color',
-  content: 'content',
-  cursor: 'cursor',
-  display: 'display',
-  fill: 'fill',
-  flex: 'flex',
-  flexBasis: 'flex-basis',
-  flexDirection: 'flex-direction',
-  flexGrow: 'flex-grow',
-  flexShrink: 'flex-shrink',
-  flexWrap: 'flex-wrap',
-  float: 'float',
-  font: 'font',
-  fontFamily: 'font-family',
-  fontSize: 'font-size',
-  fontStyle: 'font-style',
-  fontWeight: 'font-weight',
-  height: 'height',
-  justifyContent: 'justify-content',
-  left: 'left',
-  lineHeight: 'line-height',
-  listStyle: 'list-style',
-  listStyleType: 'list-style-type',
-  marginBottom: 'margin-bottom',
-  marginLeft: 'margin-left',
-  marginRight: 'margin-right',
-  marginTop: 'margin-top',
-  maxHeight: 'max-height',
-  maxWidth: 'max-width',
-  minHeight: 'min-height',
-  minWidth: 'min-width',
-  mozOsxFontSmoothing: 'moz-osx-font-smoothing',
-  objectFit: 'object-fit',
-  opacity: 'opacity',
-  order: 'order',
-  outline: 'outline',
-  outlineColor: 'outline-color',
-  outlineOffset: 'outline-offset',
-  overflow: 'overflow',
-  overflowWrap: 'overflow-wrap',
-  overflowX: 'overflow-x',
-  overflowY: 'overflow-y',
-  paddingBottom: 'padding-bottom',
-  paddingLeft: 'padding-left',
-  paddingRight: 'padding-right',
-  paddingTop: 'padding-top',
-  perspective: 'perspective',
-  pointerEvents: 'pointer-events',
-  position: 'position',
-  resize: 'resize',
-  right: 'right',
-  speak: 'speak',
-  src: 'src',
-  tableLayout: 'table-layout',
-  textAlign: 'text-align',
-  textDecoration: 'text-decoration',
-  textOverflow: 'text-overflow',
-  textTransform: 'text-transform',
-  top: 'top',
-  transform: 'transform',
-  transformOrigin: 'transform-origin',
-  transition: 'transition',
-  transitionDelay: 'transition-delay',
-  transitionDuration: 'transition-duration',
-  transitionProperty: 'transition-property',
-  transitionTimingFunction: 'transition-timing-function',
-  userSelect: 'user-select',
-  verticalAlign: 'vertical-align',
-  visibility: 'visibility',
-  webkitFontSmoothing: 'webkit-font-smoothing',
-  whiteSpace: 'white-space',
-  width: 'width',
-  wordBreak: 'word-break',
-  wordWrap: 'word-wrap',
-  zIndex: 'z-index'
-};
+function makeKebabMap(): { [key: string]: string } {
+  const kebabMap: { [key: string]: string } = {};
+  const elementStyles = (typeof document !== 'undefined' && document.createElement('div').style) || {};
+  let key: string;
+  for (key in elementStyles) {
+    if (key) {
+      kebabMap[key] = key.replace(/([A-Z])/g, '-$1').toLowerCase();
+    }
+  }
+  return kebabMap;
+}
+
+const rules: { [key: string]: string } = makeKebabMap();
```

Comment:
I don't think this is necessary. If you remove this code and leave it as a blank object, We made a change to cache things as needed. I'm going to try and tweak your PR.

Reply:
I modified the PR to just let it lazy cache. This is good enough :)